ddp-workspace/projects/ddp-dsm-ui/src/app/FON/components/register-patients-modal/register-patients-modal.component.ts

```diff
@@ -1,60 +1,113 @@
-import {Component, OnDestroy, OnInit} from '@angular/core';
+import {Component, OnDestroy, OnInit, TemplateRef, ViewChild} from '@angular/core';
 import {MatDialogRef} from '@angular/material/dialog';
 import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
-import {Subscription, tap} from 'rxjs';
+import {PatientsService} from '../../services/patients.service';
+import {forkJoin, Observable, of, Subject} from 'rxjs';
+import {HttpService} from '../../services/http.service';
+import {MatSnackBar} from '@angular/material/snack-bar';
+import {AddPatientModel} from '../../models/addPatient.model';
+import {catchError, takeUntil} from 'rxjs/operators';
+import {HttpErrorResponse} from '@angular/common/http';
 
 @Component({
   selector: 'app-add-patients-modal',
   templateUrl: './register-patients-modal.component.html',
   styleUrls: ['./register-patients-modal.component.scss'],
 })
-export class RegisterPatientsModalComponent implements OnInit {
-  uploading = false;
+export class RegisterPatientsModalComponent implements OnInit, OnDestroy {
+  isAddingPatient: boolean;
+  addedPatients: number;
 
-  readonly addPatientForm: FormGroup = this.formBuilder.group({
+  readonly patientsHttpArray: Observable<any>[] = [];
+  readonly patientsAddingForm: FormGroup = this.formBuilder.group({
```

Comment:
**_#optional_**

may be just `patiensGroup` ?

Reply:
AAlright. done